### ISSUE TITLE
Default argument for container

### ISSUE THREAD

**Issue by vreuter** [databio/pypiper#87]
It seems like a pipeline may run many--often, maybe even all--commands within a container if one is in use. Should we consider making use of a `PipelineManager`'s container the default behavior of the  `run` method? If not, is there use in continuing to have a pipeline manager continue to store a reference to a container? I like that there's ability to pass an arbitrary container each time `run` is used, but if the manager doesn't use its container for the `run` call, what does it mean for it to have a container attribute?

**Comment by nsheff** [comment]
Well, the reasoning was just to have somehwere nice to put it, like used in refgenie:

https://github.com/databio/refgenie/blob/master/src/refgenie.py

because the `get_container` function is nice to have:
https://github.com/databio/pypiper/blob/254f1e6a456b22c0c0ad2c76ed46cd9901807fe6/pypiper/manager.py#L1744-L1756

and then it just felt nice to keep that with the object, even though it's not used by default. then I can just say `pm.container`, which is standardized.

basically I think it just provides a standard location to put a container. that said, what you suggest still may be a good idea...

**Comment by nsheff** [comment]
I think I see your point more clearly now than I did last time I read this issue. I agree with you.

We can make the manager *not* have a container, and make `get_container` be a static function that returns the container ID, instead of a method on pipeline manager. agreed?

this will break refgenie, though.

**Comment by vreuter** [comment]
Either way is OK with me; at this point, I've done so much less with Docker and containerization in general that I think you're much better positioned than I am to decide about this.
